flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/tasks/CheckpointCoordinatorConfiguration.java

```diff
@@ -63,11 +65,13 @@ public CheckpointCoordinatorConfiguration(
 			int maxConcurrentCheckpoints,
 			CheckpointRetentionPolicy checkpointRetentionPolicy,
 			boolean isExactlyOnce,
-			boolean isPerfetCheckpointForRecovery) {
+			boolean isPerfetCheckpointForRecovery,
+			int tolerableCpFailureNumber) {
 
 		// sanity checks
 		if (checkpointInterval < 1 || checkpointTimeout < 1 ||
-			minPauseBetweenCheckpoints < 0 || maxConcurrentCheckpoints < 1) {
+			minPauseBetweenCheckpoints < 0 || maxConcurrentCheckpoints < 1 ||
```

Comment:
Why do we need `tolerableCpFailureNumber > Integer.MAX_VALUE`, I think an int can never bigger than `Integer.MAX_VALUE`